bindings/pyroot_experimental/pythonizations/python/ROOT/_numbadeclare.py

```diff
@@ -0,0 +1,278 @@
+    return_type = normalize_typename(return_type)
+
+    # Helper functions to determine types
+    def get_inner_type(t):
+        '''
+        Get inner typename of a templated C++ typename
+        '''
+        g = re.match('(.*)<(.*)>', t).groups(0)
+        if g and len(g) == 2:
+            return g[1]
+        raise Exception('Failed to extract template argument of type {}'.format(t))
+
+    def get_numba_type(t):
+        '''
+        Get numba type object from a C++ fundamental typename
+        '''
+        typemap = {
+                'float': nb.float32,
+                'double': nb.float64,
+                'int': nb.int32,
+                'unsigned int': nb.uint32,
+                'long': nb.int64,
+                'unsigned long': nb.uint64,
+                'bool': nb.boolean
+                }
+        if t in typemap:
+            return typemap[t]
+        raise Exception(
+                'Type {} is not supported for jitting with numba. Valid fundamental types and RVecs thereof are {}'.format(
+                    t, list(typemap.keys())))
+
+    def get_c_signature(input_types, return_type):
+        '''
+        Get C signature as numba type objects from C++ typenames
+        '''
+        c_input_types = []
+        for t in input_types:
+            if 'RVec' in t:
+                c_input_types += [nb.types.CPointer(get_numba_type(get_inner_type(t))), nb.int32]
+            else:
+                c_input_types.append(get_numba_type(t))
+        if 'RVec' in return_type:
+            # See the Python wrapper for the reason using these types
+            c_return_type = nb.void
+            c_input_types += [nb.types.CPointer(nb.int64), nb.types.CPointer(nb.int64), nb.int32]
+        else:
+            c_return_type = get_numba_type(return_type)
+        return c_return_type, c_input_types
+
+    def get_numba_signature(input_types, return_type):
+        '''
+        Get numba signature as numba type objects from C++ typenames
+        '''
+        nb_input_types = []
+        for t in input_types:
+            if 'RVec' in t:
+                nb_input_types.append(get_numba_type(get_inner_type(t))[:])
+            else:
+                nb_input_types.append(get_numba_type(t))
+        if 'RVec' in return_type:
+            nb_return_type = get_numba_type(get_inner_type(return_type))[:]
+        else:
+            nb_return_type = get_numba_type(return_type)
+        return nb_return_type, nb_input_types
+
+    # Define inner decorator without arguments
+    def inner(func, input_types=input_types, return_type=return_type, name=name):
+
+        # Jit the given Python callable with numba
+        nb_return_type, nb_input_types = get_numba_signature(input_types, return_type)
+        try:
+            nbjit = nb.jit(nb_return_type(*nb_input_types), nopython=True, inline='always')(func)
+        except:
+            raise Exception('Failed to jit Python callable {} with numba.jit'.format(func))
```

Comment:
It's good to see the pointer so you can debug whether you actually see the function you expect. Since you can replace in Python an object with a given name at any time, a pointer is the only id you have.

Reply:
:+1: